### PR TITLE
EPMDJ-2175: Actualized the drill4j ui-kit

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-2175

### CODE REVIEW

File: .storybook/storybook-components/icons-table/icons-table.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+interface Props {
+  className?: string;
+}
+
+export const IconsTable = iconsTable(({ className }: Props) => {
+  const StorybookIcons = Object.keys(Icons);
+  return (
+    <div className={className}>
+      <IconInfo>
+        {`The following npm package, @drill4j/ui-kit, includes the ${StorybookIcons.length}+ icons converted to SvgIcon components.`}
```
Comment (pavyarov) [inline .storybook/storybook-components/icons-table/icons-table.tsx]: just ${StorybookIcons.length}

File: .storybook/storybook-components/index.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export { IconsTable } from './icons-table';
+export { StoriesDecorator } from './stories-decorator';
```
Comment (pavyarov) [inline .storybook/storybook-components/index.ts]: add line

File: package.json [hunk 3]
```diff
@@ -12,9 +12,11 @@
   },
   "dependencies": {
     "@redneckz/react-bem-helper": "^2.0.4",
+    "@storybook/addon-centered": "^5.3.9",
```
Comment (pavyarov) [inline package.json]: remove

File: package.json [hunk 4]
```diff
@@ -12,9 +12,11 @@
...
     "@types/nanoid": "^2.1.0",
     "@types/react": "^16.9.17",
     "@types/react-dom": "^16.9.4",
+    "@types/storybook__addon-info": "^5.2.1",
```
Comment (pavyarov) [inline package.json]: nope

Comment (pavyarov) [reply package.json]: if this typings are required put them to devDependencies

File: package.json [hunk 5]
```diff
@@ -44,10 +47,11 @@
     "@types/react-test-renderer": "^16.9.1",
     "@typescript-eslint/eslint-plugin": "^2.16.0",
     "@typescript-eslint/parser": "^2.16.0",
-    "awesome-typescript-loader": "^5.2.1",
-    "babel-loader": "^8.0.6",
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.2",
+    "react-addons-test-utils": "^15.6.2",
```
Comment (pavyarov) [inline package.json]: remove

File: src/components/badge/__snapshots__/badge.spec.tsx.snap [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+exports[`Badge should match snapshot 1`] = `
 <span
   className="badge"
 >
   simple
 </span>
 `;
 
-exports[`should match snapshot with bold 1`] = `
+exports[`Badge should match snapshot with bold 1`] = `
```
Comment (pavyarov) [inline src/components/badge/__snapshots__/badge.spec.tsx.snap]: Bold badge...

File: src/components/icon/icons.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { IconsTable } from '../../../.storybook/storybook-components';
+
+storiesOf('Icons', module).add('Icons', () => (
```
Comment (pavyarov) [inline src/components/icon/icons.stories.tsx]: add generic-icon props

File: src/components/items-actions/items-actions.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
```
Comment (pavyarov) [inline src/components/items-actions/items-actions.spec.tsx]: remove this component, its outdated

File: src/components/menu/menu.spec.tsx [hunk 9]
```diff
@@ -1,22 +1,46 @@
...
+        },
+        {
+          label: 'Add',
+          icon: 'Edit',
+          onClick: mockFn,
+        },
+      ]}
+    />);
+    wrapper.find('element(menu-icon)').simulate('click');
+    wrapper.find('element(item)').map((node: any) => node.simulate('click'));
```
Comment (pavyarov) [inline src/components/menu/menu.spec.tsx]: fix types

File: src/components/menu/menu.stories.tsx [hunk 10]
```diff
@@ -11,6 +11,21 @@ storiesOf('menu', module).add('menu', () => (
         icon: 'Edit',
         onClick: () => {},
       },
+      {
```
Comment (pavyarov) [inline src/components/menu/menu.stories.tsx]: story name in PascalCase

Comment (pavyarov) [reply src/components/menu/menu.stories.tsx]: and menu does not show

File: src/components/menu/menu.tsx [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import React, { useState } from 'react';
```
Comment (pavyarov) [inline src/components/menu/menu.tsx]: discard this changes

File: src/components/popup/popup.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+            <h2>Popup content</h2>
+          </div>
+        </Popup>,
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should...', () => {
```
Comment (pavyarov) [inline src/components/popup/popup.spec.tsx]: should...

File: src/components/table/column.tsx [hunk 13]
```diff
@@ -0,0 +1,3 @@
+import { ColumnProps } from './table-types';
+
+export const Column = (props: ColumnProps) => null;
```
Comment (pavyarov) [inline src/components/table/column.tsx]: disable this rule at line

File: src/components/table/default-header-cell.tsx [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+}: {
+  column: ColumnProps;
+  sort?: Sort;
+  onSort?: (sortField: string) => void;
+}) => (onSort ? (
+  <TableSort onClick={() => onSort(column.name)}>
+    {column.label}
+    {sort && sort.field === column.name && (
+      <TableToggle>
+        {/* <Icons.AngleDown rotate={sort.order === 'ASC' ? 180 : undefined} height={4} width={8} /> */}
```
Comment (pavyarov) [inline src/components/table/default-header-cell.tsx]: remove this line

File: src/components/table/expandable-table/row-expander/row-expander.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+          onClick={mockFn}
+          expanded
+        />,
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should handle click', () => {
```
Comment (pavyarov) [inline src/components/table/expandable-table/row-expander/row-expander.spec.tsx]: more detailed description

File: src/components/table/table-types.ts [hunk 16]
```diff
@@ -0,0 +1,45 @@
+// @flow
```
Comment (pavyarov) [inline src/components/table/table-types.ts]: remove

File: src/components/table/table.spec.tsx [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+import renderer from 'react-test-renderer';
+
+import { Table } from './table';
+import { Column } from './column';
+
+describe('Table', () => {
+  it('should match snapshot', () => {
+    const tree = renderer
+      .create(
+        <Table data={[{
```
Comment (pavyarov) [inline src/components/table/table.spec.tsx]: remove unnecessary data

File: src/components/table/table.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+        <div>
+          <div>Total</div>
+          <div>Methods</div>
+        </div>
+      )}
+    />
+    <Column
+      name="newMethods"
+      HeaderCell={() => (
+        <div>
```
Comment (pavyarov) [inline src/components/table/table.stories.tsx]: to much div's

File: src/components/tabs/tabs.spec.tsx [hunk 19]
```diff
@@ -34,7 +30,7 @@ describe('Tabs Panel', () => {
     expect(wrapper.find('button')).toHaveLength(3);
   });
 
-  it('select a new item', () => {
+  it('should select a new item', () => {
```
Comment (pavyarov) [inline src/components/tabs/tabs.spec.tsx]: change description

File: src/components/tooltip/tooltip.spec.tsx [hunk 20]
```diff
@@ -1,16 +1,43 @@
...
         <div>Tooltip</div>
       </Tooltip>,
-    )
-    .toJSON();
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render message after mouserover', () => {
```
Comment (pavyarov) [inline src/components/tooltip/tooltip.spec.tsx]: ... at mouseover

File: src/components/tooltip/tooltip.spec.tsx [hunk 21]
```diff
@@ -1,16 +1,43 @@
...
+      message: <span>Message</span>,
+      children: <div>Tooltip</div>,
+    };
+    const wrapper = mount(<Tooltip position="bottom" {...props} />);
+    wrapper.find('div').at(1).simulate('mouseover');
+    expect(wrapper.find('element(message)').contains(<span>Message</span>)).toEqual(true);
+  });
 
-  expect(tree).toMatchSnapshot();
+  it('should hide message after mouseleave', () => {
```
Comment (pavyarov) [inline src/components/tooltip/tooltip.spec.tsx]: at

File: src/forms/inputs/checkbox/checkbox.spec.tsx [hunk 22]
```diff
@@ -0,0 +1,26 @@
...
+      <Checkbox
+        checked
+        label="checkbox"
+      />,
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('calls onChange when the checkbox label is clicked', () => {
```
Comment (pavyarov) [inline src/forms/inputs/checkbox/checkbox.spec.tsx]: should...

File: src/forms/inputs/input/input.module.scss [hunk 23]
```diff
@@ -32,6 +32,10 @@
       opacity: 0.5;
     }
 
+    &--checked {
```
Comment (pavyarov) [inline src/forms/inputs/input/input.module.scss]: check ReactFinalForm field

File: src/forms/inputs/input/input.spec.tsx [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+    expect(mockFn).toHaveBeenCalled();
+    expect(wrapper.prop('value')).toContain('input component');
+    expect(wrapper.prop('placeholder')).toContain('placeholder initial state');
+  });
+
+  it('should rerender component if the state has changed ', () => {
+    const wrapper = mount(
+      <Input {...props} />,
+    );
+    expect(wrapper.find('input').hasClass('input-wrapper__input-element--rounded')).toEqual(false);
```
Comment (pavyarov) [inline src/forms/inputs/input/input.spec.tsx]: this looks like bem helper tests

File: src/forms/inputs/toggler/toggler.spec.tsx [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+        <Toggler
+          label="toggler"
+        />,
+      )
+      .toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render component with initial state', () => {
```
Comment (pavyarov) [inline src/forms/inputs/toggler/toggler.spec.tsx]: you have snapshot test for this

File: src/forms/inputs/toggler/toggler.spec.tsx [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+      value: false,
+    };
+    const wrapper = mount(
+      <Toggler {...props} />,
+    );
+    wrapper.find('input').hasClass('input-wrapper__input-element');
+    expect(wrapper.find('input').hasClass('input-wrapper__input-element--checked')).toEqual(false);
+    expect(wrapper.props().value).toEqual(false);
+    wrapper.setProps({ value: true });
+    wrapper.update();
```
Comment (pavyarov) [inline src/forms/inputs/toggler/toggler.spec.tsx]: ?

File: src/layouts/panel/panel.spec.tsx [hunk 27]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+
+import { Panel } from './panel';
+
+describe('Panel', () => {
+  it('should match snapshot', () => {
+    const tree = renderer.create(
+      <Panel>
```
Comment (pavyarov) [inline src/layouts/panel/panel.spec.tsx]: more usable examples

File: src/layouts/panel/panel.stories.tsx [hunk 28]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+
+import { Panel } from './panel';
+
+storiesOf('Panel', module).add('Panel', () => (
```
Comment (pavyarov) [inline src/layouts/panel/panel.stories.tsx]: more usable cases

File: src/components/icon/icons.stories.tsx (hunk 7 above)
Comment (RomanAchilov) [reply src/components/icon/icons.stories.tsx]: I have a small question

File: src/components/menu/menu.spec.tsx (hunk 9 above)
Comment (RomanAchilov) [reply src/components/menu/menu.spec.tsx]: just delete any

File: .storybook/storybook-components/icons-table/index.ts [hunk 29]
```diff
@@ -0,0 +1,2 @@
+export { IconsTable } from './icons-table';
+
```
Comment (pavyarov) [inline .storybook/storybook-components/icons-table/index.ts]: somethink wrong, check for empty spaces

File: .storybook/storybook-components/stories-decorator/index.ts [hunk 30]
```diff
@@ -0,0 +1,2 @@
+export {StoriesDecorator} from './stories-decorator';
```
Comment (pavyarov) [inline .storybook/storybook-components/stories-decorator/index.ts]: same

File: src/components/badge/__snapshots__/badge.spec.tsx.snap [hunk 31]
```diff
@@ -1,14 +1,14 @@
...
+exports[`Badge should match snapshot 1`] = `
 <span
   className="badge"
 >
   simple
 </span>
 `;
 
-exports[`should match snapshot with bold 1`] = `
+exports[`Badge should match snapshot bold badge 1`] = `
```
Comment (pavyarov) [inline src/components/badge/__snapshots__/badge.spec.tsx.snap]: "Badge should match snapshot bold badge" sound strange, rename

File: src/components/tabs/tabs.spec.tsx [hunk 32]
```diff
@@ -34,7 +30,7 @@ describe('Tabs Panel', () => {
     expect(wrapper.find('button')).toHaveLength(3);
   });
 
-  it('select a new item', () => {
+  it('should chancge active tab after click', () => {
```
Comment (pavyarov) [inline src/components/tabs/tabs.spec.tsx]: chancge?

File: src/forms/inputs/input/input.module.scss [hunk 33]
```diff
@@ -32,6 +31,10 @@
       opacity: 0.5;
     }
 
+    &--success {
```
Comment (pavyarov) [inline src/forms/inputs/input/input.module.scss]: PLEASE check prop name at ReactFinalProps

File: src/components/table/selectable-table.spec.tsx [hunk 34]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { mount } from 'enzyme';
```
Comment (pavyarov) [inline src/components/table/selectable-table.spec.tsx]: remove unused imports

**Pull request merged**